src/libstd/macros.rs

```diff
@@ -19,11 +20,24 @@ macro_rules! panic {
         $crate::panic!($msg)
     });
     ($fmt:expr, $($arg:tt)+) => ({
-        $crate::rt::begin_panic_fmt(&$crate::format_args!($fmt, $($arg)+),
-                                    &($crate::file!(), $crate::line!(), $crate::column!()))
+        $crate::rt::begin_panic_fmt(&$crate::format_args!($fmt, $($arg)+))
     });
 }
 
+#[cfg(not(bootstrap))]
+#[doc(include = "../libcore/macros/panic.md")]
+#[macro_export]
+#[stable(feature = "rust1", since = "1.0.0")]
+#[allow_internal_unstable(libstd_sys_internals)]
+macro_rules! panic {
+    () => {{ $crate::panic!("explicit panic") }};
+    ($msg:expr) => {{ $crate::rt::begin_panic($msg) }};
+    ($msg:expr,) => {{ $crate::panic!($msg) }};
+    ($fmt:expr, $($arg:tt)+) => {{
+        $crate::rt::begin_panic_fmt(&$crate::format_args!($fmt, $($arg)+))
+    }};
```

Comment:
You've changed parens to braces which makes me realize that the braces in the original were unnecessary (presumably panicking needed a block in the past?), and you can get rid of them here if you want.

Reply:
There are tests that are sensitive to macro expanding to a block vs a bare expr. Want me to figure out which ones it was?

Reply:
```
failures:

---- [ui] ui/block-result/issue-5500.rs stdout ----
diff of stderr:

4	LL | fn main() {
5	   |           - expected `()` because of default return type
6	LL |     &panic!()
-	   |     ^^^^^^^^^
-	   |     |
-	   |     expected `()`, found reference
-	   |     help: consider removing the borrow: `panic!()`
+	   |     ^^^^^^^^^ expected `()`, found `&!`
11	   |
12	   = note: expected unit type `()`
-	              found reference `&_`
+	              found reference `&!`
14	
15	error: aborting due to previous error
16	


The actual stderr differed from the expected stderr.
Actual stderr saved to /home/adam/code/rust/build/x86_64-unknown-linux-gnu/test/ui/block-result/issue-5500/issue-5500.stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args block-result/issue-5500.rs`

error: 1 errors occurred comparing output.
status: exit code: 1
command: "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/home/adam/code/rust/src/test/ui/block-result/issue-5500.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/test/ui/block-result/issue-5500" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/home/adam/code/rust/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/test/ui/block-result/issue-5500/auxiliary" "-A" "unused"
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------
error[E0308]: mismatched types
  --> /home/adam/code/rust/src/test/ui/block-result/issue-5500.rs:2:5
   |
LL | fn main() {
   |           - expected `()` because of default return type
LL |     &panic!()
   |     ^^^^^^^^^ expected `()`, found `&!`
   |
   = note: expected unit type `()`
              found reference `&!`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.

------------------------------------------


---- [ui] ui/diverging-tuple-parts-39485.rs stdout ----
diff of stderr:

1	error[E0308]: mismatched types
2	  --> $DIR/diverging-tuple-parts-39485.rs:8:5
3	   |
+	LL | fn g() {
+	   |        - help: try adding a return type: `-> &!`
4	LL |     &panic!()
-	   |     ^^^^^^^^^ expected `()`, found reference
+	   |     ^^^^^^^^^ expected `()`, found `&!`
6	   |
7	   = note: expected unit type `()`
-	              found reference `&_`
-	help: try adding a return type
-	   |
-	LL | fn g() -> &_ {
-	   |        ^^^^^
-	help: consider removing the borrow
-	   |
-	LL |     panic!()
-	   |     ^^^^^^^^
+	              found reference `&!`
17	
18	error[E0308]: mismatched types
19	  --> $DIR/diverging-tuple-parts-39485.rs:12:5


The actual stderr differed from the expected stderr.
Actual stderr saved to /home/adam/code/rust/build/x86_64-unknown-linux-gnu/test/ui/diverging-tuple-parts-39485/diverging-tuple-parts-39485.stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args diverging-tuple-parts-39485.rs`

error: 1 errors occurred comparing output.
status: exit code: 1
command: "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/home/adam/code/rust/src/test/ui/diverging-tuple-parts-39485.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/test/ui/diverging-tuple-parts-39485" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/home/adam/code/rust/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/test/ui/diverging-tuple-parts-39485/auxiliary" "-A" "unused"
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------
error[E0308]: mismatched types
  --> /home/adam/code/rust/src/test/ui/diverging-tuple-parts-39485.rs:8:5
   |
LL | fn g() {
   |        - help: try adding a return type: `-> &!`
LL |     &panic!() //~ ERROR mismatched types
   |     ^^^^^^^^^ expected `()`, found `&!`
   |
   = note: expected unit type `()`
              found reference `&!`

error[E0308]: mismatched types
  --> /home/adam/code/rust/src/test/ui/diverging-tuple-parts-39485.rs:12:5
   |
LL | fn f() -> isize {
   |           ----- expected `isize` because of return type
LL |     (return 1, return 2) //~ ERROR mismatched types
   |     ^^^^^^^^^^^^^^^^^^^^ expected `isize`, found tuple
   |
   = note: expected type `isize`
             found tuple `(!, !)`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.

------------------------------------------


---- [ui] ui/issues/issue-5500-1.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit code: 1
command: "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/home/adam/code/rust/src/test/ui/issues/issue-5500-1.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-5500-1" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/home/adam/code/rust/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/home/adam/code/rust/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-5500-1/auxiliary" "-A" "unused"
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------
error[E0308]: mismatched types
  --> /home/adam/code/rust/src/test/ui/issues/issue-5500-1.rs:14:18
   |
LL |     _iter.node = &panic!()
   |                  ^^^^^^^^^ expected `usize`, found `!`
   |
   = note: expected reference `&usize`
              found reference `&!`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.

------------------------------------------



failures:
    [ui] ui/block-result/issue-5500.rs
    [ui] ui/diverging-tuple-parts-39485.rs
    [ui] ui/issues/issue-5500-1.rs

test result: FAILED. 9420 passed; 3 failed; 45 ignored; 0 measured; 0 filtered out
```